src/flamenco/capture/solcap.md

```diff
@@ -0,0 +1,256 @@
+[Enhanced Packet Block (EPB) #2]    ← Second message
+...
+[Enhanced Packet Block (EPB) #N]    ← Nth message
+```
+
+The Section Header Block and Interface Description Block are written once at the beginning of the file. All subsequent messages are wrapped in Enhanced Packet Blocks.
+
+### Layer 2: Transport Layer - Enhanced Packet Blocks
+
+Each solcap message is encapsulated in a PCapNG Enhanced Packet Block (EPB). The EPB provides:
+- Block type identification
+- Total block length (for forward seeking)
+- Timestamp information
+- Packet length (captured vs original)
+- Redundant length footer (for backward seeking)
+
+These fields are not directly read in solcap files, and provide no use
+
+**Note**: Any Simple Packet Blocks are disregarded
+
+### Layer 3: Muxing Layer - Message Type Identification
+
+The Internal Chunk Header immediately follows the EPB header within each packet. It serves as the **primary muxing layer** that identifies which type of solcap message follows. The `block_type` field is the discriminator that allows parsers to correctly decode the message payload.
+
+This layer also provides:
+- **Temporal context**: The `slot` field indicates which slot this message relates to
+- **Ordering**: The `txn_idx` field provides transaction-level ordering within a slot
+- **Type routing**: The `block_type` field routes to the appropriate message parser
+
+### Layer 4: Message Layer - Payload Data
+
+The message payload immediately follows the Internal Chunk Header. The format is determined by the `block_type` from Layer 3. Currently defined message types include:
+
+- **Account Updates** (`SOLCAP_WRITE_ACCOUNT_HDR` + `SOLCAP_WRITE_ACCOUNT_DATA`)
+- **Bank Preimages** (`SOLCAP_WRITE_BANK_PREIMAGE`)
+
+### Block Alignment
+
+All blocks must be aligned to 4-byte boundaries. Padding bytes (zeros) are inserted after the message payload if needed to achieve this alignment.
+
+---
+
+## Message Types
+
+Solcap defines the following message types for capturing validator runtime events:
+
+### Account State Changes
+
+**Purpose**: Capture all modifications to account state during transaction execution.
+
+- `SOLCAP_WRITE_ACCOUNT_HDR` (1): Account update header
+  - Contains account pubkey, metadata, and data length
+  - Must be followed by `SOLCAP_WRITE_ACCOUNT_DATA`
+
+- `SOLCAP_WRITE_ACCOUNT_DATA` (2): Account data payload
+  - Contains the actual account data bytes
+  - Follows immediately after `SOLCAP_WRITE_ACCOUNT_HDR`
+
+### Bank State
+
+**Purpose**: Capture the complete state of a bank (block) after processing.
+
+- `SOLCAP_WRITE_BANK_PREIMAGE` (6): Bank preimage data
+  - Contains bank hash, previous bank hash, accounts hash, PoH hash
+  - Includes transaction count for the slot
+  - Enables verification of deterministic execution
+
+---
+
+## Data Structures
+
+All structures use packed layout (`__attribute__((packed))`) to ensure consistent binary representation across platforms.
```

Comment:
Packed layout is a C specific concept and is already implied by nature of being a spec, remove this section. Would mention endianness though